src/Illuminate/Support/Siesta.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Carbon\CarbonInterval;
+use DateInterval;
+use PHPUnit\Framework\Assert as PHPUnit;
+use RuntimeException;
+
+class Siesta
+{
+    /**
+     * The total duration to sleep.
+     *
+     * @var \Carbon\CarbonInterval
+     */
+    public $duration;
```

Comment:
I know, that it isn't supposed to use Siesta this way, but let's assume this example:
```
function ImAFunOfPrototypeDesignPattern(Siesta $pause)
{
    dump(time());
    (clone $pause)->and(3)->seconds(); // waits 5 seconds as expected.
    dump(time());
    clone $pause; // also waits for 5 seconds instead of 2.
    dump(time());
    // also waits for 5 seconds instead of 2.
    // who doesn't want delay here can write $pause->duration = CarbonInterval::seconds(0); it's ok.
    // or $pause->discard() to also be used in tests.
}

dump(time());
ImAFunOfPrototypeDesignPattern(Siesta::for(2)->seconds());
dump(time());
```
Should the `__clone()` method also be implemented?

Reply:
I don't think we need to consider this. That is not the intended use of this class.